cores/esp8266/Updater.cpp

```diff
@@ -223,18 +223,26 @@ bool UpdaterClass::end(bool evenIfRemaining){
 
 bool UpdaterClass::_writeBuffer(){
 
-  bool result = true;
+  bool eraseResult = true;
+  bool writeResult = true;
   if (_currentAddress % FLASH_SECTOR_SIZE == 0) {
     if(!_async) yield();
-    result = ESP.flashEraseSector(_currentAddress/FLASH_SECTOR_SIZE);
+    eraseResult = ESP.flashEraseSector(_currentAddress/FLASH_SECTOR_SIZE);
   }
   
-  if (result) {
+  if (eraseResult) {
     if(!_async) yield();
-    result = ESP.flashWrite(_currentAddress, (uint32_t*) _buffer, _bufferLen);
+    writeResult = ESP.flashWrite(_currentAddress, (uint32_t*) _buffer, _bufferLen);
+  } else { // if erase was not successful
+    _error = UPDATE_ERROR_ERASE;
```

Comment:
These three lines are repeated all over the place, and your change adds another repeat:
```
    _error = some_error;
#ifdef DEBUG_UPDATER
    printError(DEBUG_UPDATER);
#endif

```
Could you please create a new private method "setError(int err)" that wraps the three lines, and replace all cases in the file where the three lines appear by a call to the method? In this case, it would look like this:
```
_currentAddress = (_startAddress + _size);
setError(UPDATE_ERROR_ERASE);
return false;
```
There are a few cases where you'll need to swap lines (like in your change), and other cases where the #ifdef is even missing.

Reply:
I believe I have fixed what you have mentioned, please check the latest commits. 

Thanks for the heads up.